### DISCUSSION TITLE
Psalm not show attributes methods errors in another file

### DISCUSSION THREAD

**Discussion by vadikgr-hub** [vimeo/psalm#7888] [Psalm development (AKA internals)]
If I call the method \Test\MyClass::test('string'); in file 1. psalm work correct and show error "psalm: InvalidScalarArgument: Argument 1 of..."
Why in file 2 psalm dont show attributes errors(InvalidScalarArgument)?

File 1:
```
<?php
//file1.php
namespace Test;

class MyClass{
    /**
     * @psalm-param int $a
     * @return void
     */
    static function test(int $a){}
}
```
File 2:
```
<?php
//file2.php
    \Test\MyClass::test('string');
```
psalm.xml:
```
<?xml version="1.0"?>

<psalm
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xmlns="https://getpsalm.org/schema/config"
        xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
        xmlns:xi="http://www.w3.org/2001/XInclude"
        errorLevel="1"
>
</psalm>
```

Please explain to a newbie what I'm doing wrong?

**Comment by AndrolGenhald** [comment]
How are your classes loaded? Are you using PSR-4 autoloading with composer?

**Reply by vadikgr-hub** [reply]
Thanks for the tip and the answer. No, our module with classes is connected inside the framework, not in the composer.

PhpStorm sees and inspects our classes correctly, but the psalm does not (in file 2).

If include require_once('full path to script'); in file 2 then all works correct. But it is no good idea.
Also, our composer is not connected via an explicit path, but like this - require_once($documentRoot . '/local/include/vendor/autoload.php');

Are there any other ways to tell the psalm where to look for these classes?

**Reply by AndrolGenhald** [reply]
You can try using the [`autoloader` configuration](https://psalm.dev/docs/running_psalm/configuration/#autoloader), but I've always used composer so I'm not too familiar with how it works myself.
